### PR TITLE
Lazy load view components in MapView.

### LINKED ISSUE #76: Lazy initialize views
When creating a `mapView`, we inflate all overlain children at once even if the user has configured the map to not shown them (eg. through `MapboxMapOptions` or xml attributes). We have received a request to only lazy initialize these components when needed.

### PR DESCRIPTION
**Pull request by pengdev**
Resolves #76 .
This PR will introduce lazy loading of view components (compass view, logo view and attribution view) of the map.

### CODE REVIEW

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/UiSettings.java [hunk 1]
```diff
@@ -224,21 +256,31 @@ private void saveLogo(Bundle outState) {
...
+    isAttributionInitialized = true;
+    setAttributionImage();
+    mapView.initialiseAttributionHandlers();
     setAttributionEnabled(options.getAttributionEnabled());
     setAttributionGravity(options.getAttributionGravity());
     setAttributionMargins(context, options.getAttributionMargins());
     int attributionTintColor = options.getAttributionTintColor();
     setAttributionTintColor(attributionTintColor != -1
       ? attributionTintColor : ColorUtils.getPrimaryColor(context));
+    Logger.d("uisettings", "initialiseAttribution");
```
Comment (tobrun) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/UiSettings.java]: debug logging?

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/UiSettings.java [hunk 2]
```diff
@@ -224,21 +256,31 @@ private void saveLogo(Bundle outState) {
...
+      setLogoImage();
+      isLogoInitialized = true;
+    }
     setLogoEnabled(savedInstanceState.getBoolean(MapboxConstants.STATE_LOGO_ENABLED));
     setLogoGravity(savedInstanceState.getInt(MapboxConstants.STATE_LOGO_GRAVITY));
     setLogoMargins(savedInstanceState.getInt(MapboxConstants.STATE_LOGO_MARGIN_LEFT),
       savedInstanceState.getInt(MapboxConstants.STATE_LOGO_MARGIN_TOP),
       savedInstanceState.getInt(MapboxConstants.STATE_LOGO_MARGIN_RIGHT),
       savedInstanceState.getInt(MapboxConstants.STATE_LOGO_MARGIN_BOTTOM));
+    Logger.d("uisettings", "restoreLogo");
```
Comment (tobrun) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/UiSettings.java]: debug logging?

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java [hunk 3]
```diff
@@ -135,9 +142,7 @@ protected void initialize(@NonNull final Context context, @NonNull final MapboxM
     View view = LayoutInflater.from(context).inflate(R.layout.mapbox_mapview_internal, this);
```
Comment (tobrun) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java]: Lazy loading would also mean lazy inflating. We would probably need to remove `mapbox_mapview_interal` an only create/add views when they a requested.

Comment (pengdev) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java]: Good point, adjusted to lazy inflate view components.

**Comment by pengdev** [comment]
@tobrun ready for another round of 👀

**Pull request merged**